### PR TITLE
Add learning path doc

### PR DESCRIPTION
**Pull request by SaxenaKaustubh**

### CODE REVIEW

File: docs/learning_plan.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+* JavaScript: 
+    * Extensivley used with the team
+    * Currently, at version ES6
+    * Team uses Babel 6 (will move to Babel 7 soon)
+* Swift
+    * Used to build native components
+* Objective C
+    * Was used in the past to build native components
+    * Team is slowly moving away from Objective C
+* React
```
Comment (laynemoseley) [inline docs/learning_plan.md]: I would make this section titled `React Native`. We don't use any vanilla `React`.

Comment (laynemoseley) [reply docs/learning_plan.md]: or, maybe remove this completely

File: docs/learning_plan.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    * Used for state management within the React app
+    * Integrated with the networking stack
+    * Team is moving away from Redux because of maintenance complexities 
+* fetchPropsFor
+    * Internally developed middleware solution sitting between React and networking components
+* React Native
+    * Used to create the apps
+    * Important components:
+        * Stylesheets: used for styling - duh!
+        * Flexbox: used for wrapping componenets
```
Comment (laynemoseley) [inline docs/learning_plan.md]: `Flexbox: Used for layout of React Native components.` (this is part of StyleSheet)

File: docs/learning_plan.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        * Format message: used for localization
+* Flow
+    * Used for typing
```
Comment (laynemoseley) [inline docs/learning_plan.md]: `Type Checking` (https://flow.org)

File: docs/learning_plan.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+* Jest
```
Comment (laynemoseley) [inline docs/learning_plan.md]: https://jestjs.io

File: docs/learning_plan.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    * Unit testing framework
+* Lodash
```
Comment (laynemoseley) [inline docs/learning_plan.md]: https://lodash.com

**Comment by bootstraponline** [comment]
I pushed a commit that cleans up the doc a bit, let me know what you think. I agree with the suggestions from Layne. Linking to the different tech will be helpful.

**Comment by bootstraponline** [comment]
Manuel mentioned this resource:

> There is a react native sample app that allows you to test all the UI components (https://github.com/facebook/react-native/tree/master/RNTester).

**Pull request merged**